plugins/plugin_utils/_hashi_vault_lookup_base.py

```diff
@@ -42,6 +42,13 @@ def parse_kev_term(self, term, plugin_name, first_unqualified=None):
                 else:
                     raise AnsibleError("%s lookup plugin needs key=value pairs, but received %s" % (plugin_name, term))
 
+            if key in param_dict:
+                removed_in = '5.0.0'
+                msg = "Duplicate key '%s' in the term string '%s'." % (key, term)
+                display.deprecated(msg + "\nIn version %s of the collection, this will raise an exception." % (removed_in, ), removed_in)
```

Comment:
I can't figure out why use this complex form `(removed_in,)`. `% removed_id` works the same why, doesn't it? I'd use this variant if you don't want to use f-string:
```python
deprecated("%s\nIn version %s of the collection, this will raise an exception." % (msg, removed_in), removed_in)
```
f-string variant:
```python
display.deprecated(f"{msg}\nIn version {removed_in} of the collection, this will raise an exception.", removed_in)
```

Reply:
I got into the habit of using a tuple with a single value `(removed_in,)` because if `removed_in` happened to itself be a tuple, then the `% removed_in` form would fail if its contents don't match the number of templated items. 

It's mostly guarding against variables that could have different types; it's not important here I guess, since we're setting the variable explicitly, but it's a thing I try to always do anyway for consistency.

The reason we haven't used f-strings historically is that we had to support Python 2.7/3.5. That's no longer the case, but I keep using `%` strings for consistency.

I might look to do a bulk update at a later time, but not sure I want to start introducing different methods one at a time.

---

All that aside, this call will be gone in a few months for v5 😝